pkg/vault/client_test.go

```diff
@@ -9,19 +9,35 @@ import (
 	hashivault "github.com/hashicorp/vault/vault"
 )
 
-type MockAuthClient struct{}
+type GitHubMockAuthClient struct{}
+type GoogleMockAuthClient struct{}
 
-func (m *MockAuthClient) GetVaultToken(c *vaultClient) (string, error) { return "token", nil }
+func (m *GoogleMockAuthClient) GetVaultToken(gc *vaultClient) (string, error)  { return "token", nil }
+func (m *GitHubMockAuthClient) GetVaultToken(ghc *vaultClient) (string, error) { return "token", nil }
 
-var authClient AuthClient
-var c *config
+var googleAuthClient AuthClient
+var githubAuthClient AuthClient
+var gc *config
+var ghc *config
 var ctx context.Context
 var secretKey string
 var secretValue string
 
 func init() {
-	c = &config{
+	gc = &config{
 		project:        "test",
+		googleAuth:     true,
+		serviceAccount: "none",
+		traceEnabled:   false,
+		tracePrefix:    "test",
+		vaultRole:      "read",
+	}
+
+	ghc = &config{
+		project:        "test",
+		googleAuth:     false,
+		githubAuth:     true,
```

Comment:
wonder if there could be an `authMethod` sort of variable here instead of multiple true/false flags. its good it works but doesn't feel like it'd scale well with more auth methods and feels wrong having to set false for things instead of just true for the one you want / it being technically possible to set multiple things to true